src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -359,6 +390,49 @@ internal void OnCompilationEnd(CompilationAnalysisContext context)
                 }
             }
 
+            private void ProcessTypeForwardedAttributes(Compilation compilation, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
+            {
+                var typeForwardedToAttribute = compilation.GetTypeByMetadataName("System.Runtime.CompilerServices.TypeForwardedToAttribute");
+
+                if (typeForwardedToAttribute != null)
+                {
+                    foreach (var attribute in compilation.Assembly.GetAttributes())
+                    {
+                        if (typeForwardedToAttribute.Equals(attribute.AttributeClass))
+                        {
+                            var forwardedType = attribute.ConstructorArguments.Single().Value as INamedTypeSymbol;
```

Comment:
@AlekseyTs I'm fixing up the dodgy signature assumption, but the type equality pattern is what we use across the IDE. I don't think there's an easy way to ask "is this type's metadata name X?" because #1891 points out we still have no way to get the full metadata name. I could walk the namespace parents and check all those, but that seems overkill.